src/topojson/topojson.ml

```diff
@@ -222,34 +228,41 @@ module Make (J : Intf.Json) = struct
       | MultiPolygon of MultiPolygon.t
       | Collection of t list
 
-    and t = geometry * (string * json) list
+    and t = {
+      geometry : geometry;
+      properties : (string * json) list;
+      foreign_members : (string * json) list;
+    }
 
     let rec base_of_json json =
       let fm = foreign_members json in
+      let props t = t.properties in
       match J.find json [ "type" ] with
       | Some typ -> (
           match J.to_string typ with
           | Ok "Point" ->
-              Result.map (fun v -> (Point v, fm)) @@ Point.base_of_json json
+              Result.map (fun v -> (Point v, fm, props))
+              @@ Point.base_of_json json
           | Ok "MultiPoint" ->
-              Result.map (fun v -> (MultiPoint v, fm))
+              Result.map (fun v -> (MultiPoint v, fm, props))
```

Comment:
Same thing applies here, you are constructing a triple when you need to be constructing the record.